Add ability to hide menu bar and dock item

jessepollak opened the pull request:
This PR adds a settings section where users can configure whether the menu bar item shows.



![screenshot 2017-02-18 19 34 27](https://cloud.githubusercontent.com/assets/1097953/23098037/baaf2ecc-f611-11e6-8117-5c8cfc5c87b3.png)
![screenshot 2017-02-18 19 34 23](https://cloud.githubusercontent.com/assets/1097953/23098038/bab11e4e-f611-11e6-92ad-e2a83551448a.png)

jessepollak committed:
feat(settings): add support for hiding dock item and launching on startup

jessepollak commented:
I implemented this to solve #34 but just realized that issue actually hopes to achieve the opposite -- hiding the dock item, not the menu bar. This may still be a nice configuration option to have!

jessepollak commented:
Just added support for hiding the dock item as well! One thing that's worth considering is that our settings section is getting a little crowded -- @jenbennings any thoughts on that?

![screenshot 2017-02-18 19 55 56](https://cloud.githubusercontent.com/assets/1097953/23098127/49152c8c-f614-11e6-86d1-39076aa2bb58.png)

jenbennings commented:
This is awesome, I've been wanting this for ages!

Agree on settings starting to get a bit busy. We should be able to condense these into a smaller radio button style. I can jump on this PR today/tomorrow.

jenbennings commented:
Thinking through this a bit more—the current settings imply there's a state where both the menu _and_ dock icons can be hidden. Does that even work? 🤔 

We probably need to be more opinionated here. I feel like taskbar icon as a default is fine, with the ability to hide the dock icon, is the way to go. Thoughts @jorilallo @maksim-s?

<img width="682" alt="screen shot 2017-02-18 at 7 18 34 pm" src="https://cloud.githubusercontent.com/assets/3468612/23098905/1359189c-f60f-11e6-9475-ad9013b9fb5e.png">

This would cover us for #34 I imagine.

jessepollak commented:
@jenbennings OK, I just:

* Removed option to disable the menu bar item
* Updated the UI to have a macOS preferences section matching your spec
* Added support for auto launching Lionshare at startup (set to `false` by default)
* Refactored the settings view logic to create a single `CheckboxSetting` that is used across the macOS preferences and the asset list

Open for any and all feedback!

![screenshot 2017-02-19 13 41 54](https://cloud.githubusercontent.com/assets/1097953/23105231/a45d1880-f6a9-11e6-8cb8-76750d84da77.png)

jorilallo commented on src/stores/index.js:
```diff
@@ -1,4 +1,4 @@
-import { ipcRenderer } from 'electron';
+import { ipcRenderer, ipcMain } from 'electron';
```
not used

jorilallo commented:
Looks great! one tiny nit and rebase, we merge after :)

jessepollak commented:
@jorilallo updated and rebased (and did a little refactor on loading settings from JSON) - let me know if this works :)

jorilallo commented:
@jessepollak looks good, thanks 🙂
